### PR TITLE
feat: admin update product

### PR DESCRIPTION
**Pull request by Demandtech**
## What does this PR do?

This pr allow super admin to update product status
## Checklist:

- [x] Add status column to product table
- [x] Allow only super admin to change product status
- [x] Throw forbidden error when other users call the endpoint
- [x] Added Unit test to services
## Response
success
```
{
   "message": "Product status updated successfully",
   "status_code": 200,
   "data":{
       "id": "19cded37-05a4-48dc-9cf2-632f8a253cba",
       "created_at": "2024-07-30T02:19:35.254Z",
       "updated_at": "2024-07-30T02:21:57.126Z",
       "name": "test name",
       "description": "test description",
       "quantity": 20,
       "price": 10000,
       "status": "draft",
       "image": "dddd",
       "stock_status": "in stock"
   }
}
```
Forbidden response
```
{
    "error": "Forbidden",
    "status_code": 403,
    "message": "Forbidden user"
}
```
## Test screenshot
Screenshot 2024-07-30 at 10 49 44 AM

### CODE REVIEW

**Comment by Heba-WebDev** [comment]
Update the PR title please, it should be feat: <name>

File: src/modules/products/entities/product.entity.ts
Comment (Heba-WebDev) [inline src/modules/products/entities/product.entity.ts:1]: If an import is not needed, remove it. Don't commented out like this please.

Comment (Heba-WebDev) [inline src/modules/products/entities/product.entity.ts:1]: Remove the commented out imports and remove the user relationship.

File: src/modules/products/products.module.ts
Comment (Heba-WebDev) [inline src/modules/products/products.module.ts:1]: Again, out commented code should be removed

File: src/modules/products/products.controller.ts
Comment (Heba-WebDev) [inline src/modules/products/products.controller.ts:1]: You're not using  @UseGuards(OwnershipGuard) on your endpoints, why? Add it please

Comment (TheCodeGhinux) [reply src/modules/products/products.controller.ts:1]: owner guard is for organisation. I'm working on the user guard now.

Comment (Heba-WebDev) [reply src/modules/products/products.controller.ts:1]: @TheCodeGhinux The owner of an org is the one that can sell and thus can change the status of a product

**Review by Heba-WebDev — APPROVED** [review]

**Pull request closed without merging**